source/rock/middle/FunctionCall.ooc

```diff
@@ -1255,6 +1259,32 @@ FunctionCall: class extends Expression {
                                     return null
                                 }
                             }
+                        } else if (retType) {
+                            if (implArg instanceOf?(FunctionDecl)) {
```

Comment:
I feel weird having a match with a single branch, but I guess I'll get used to it